Implement tyeps on client-side

nabijaczleweli opened the pull request:
Should we stay with `browserify` or use a module system?
If `browserify`, should we force-include `victor` and `keypress`?

EthanSteinberg commented on client/src/game.ts:
```diff
@@ -0,0 +1,231 @@
+    // DOM events
+    //
+
+    onDOMReady = (): void => {
```
Minor annoyance, these should probably be methods.

nabijaczleweli replied:
nah, scoping is a bitch, see https://github.com/LoungeCPP/Tatsoryk/pull/44/commits/263a259232f72604256403c052beba2f0c2c1042

EthanSteinberg commented on client/src/gamescreen.ts:
```diff
@@ -0,0 +1,124 @@
+    // For aim vector rendering.
+    mousePos: MousePos = null;
+
+    keymap: Array<boolean> = [];
```
Key map should be an Object or Map, not an Array but it doesn't really matter.

nabijaczleweli replied:
Done

EthanSteinberg commented:
It looks good. I only had two minor comments. :+1: Nice work.

nabijaczleweli committed:
Implement client in TS. Add client to Travis

Add client README
Add client Makefile

Add node package def and typings definitions
Use Browserify

Remove heir and EventEmitter
`heir` is unneeded because we're using an acutal language
EventEmitter is npm-dep "wolfy87-eventemitter"
Other dependencies are "ambient", a.k.a. they just inject into the
browser and we utilise them from there
